operator/v1/types_ingress.go

```diff
@@ -1054,6 +1054,22 @@ type ContainerLoggingDestinationParameters struct {
 }
 
 // LoggingDestination describes a destination for log messages.
+// + ---
+// + We added the spec.logging.access.destination.container and
+// + spec.logging.access.destination.syslog fields in OpenShift 4.5 without any
+// + validation rules to prevent the user from specifying both fields at the
+// + same time.  Adding such validation rules in a subsequent release would
+// + break API compatibility between releases.
+// +
+// + We added the spec.logging.access.destination.container.maxLength field in
+// + OpenShift 4.14, and we added the following validation rules in the same
+// + release, so they do not break compatibility between releases.  Note that
+// + API defaulting requires us to allow the default value for
+// + container.maxLength if container is non-null even when type is Syslog.
+// +
+// +kubebuilder:validation:XValidation:rule=`!has(self.type) || self.type == "Container" || !has(self.container) || !has(self.container.maxLength) || self.container.maxLength == 1024`,message="container cannot be specified unless type is Container"
```

Comment:
It seems this doesn't behave properly if `type` is missing.

Reply:
I'll add a "Should require spec.logging.access.destination.type if spec.logging.access is not null" test case and observe what happens.

Reply:
We do indeed have a validation error when `type` is missing.  See <https://github.com/openshift/api/compare/484706adab46842422924bc7ab8e01e4fb477666..66af0ebd9446ee5d5464c1b8c615d87fa11622c1>.